File: app/src/main/java/com/rockon999/android/leanbacklauncher/apps/AppCategory.java
package com.rockon999.android.leanbacklauncher.apps;

import android.util.SparseArray;

public enum AppCategory {
    OTHER(0),
    SETTINGS(1),
    VIDEO(2),
    MUSIC(3),
    GAME(4);

    /**
     * Created by rockon999 on 2/18/18.
     */
    public final int code;

    AppCategory(int code) {
        this.code = code;
    }

    private static final SparseArray<AppCategory> intToType = new SparseArray<>();

    public static AppCategory fromCategoryCode(int code) {
        AppCategory category = intToType.get(code);
        return category == null ? AppCategory.OTHER : category;
    }

    public static AppCategory fromName(String name) {
        if (name == null) return null;
        return valueOf(name.toUpperCase().trim());
    }

    static {
        for (AppCategory type : values()) {
            intToType.put(type.code, type);
        }
    }
}

File: app/src/main/java/com/amazon/tv/firetv/leanbacklauncher/apps/AppCategory.kt
package com.amazon.tv.firetv.leanbacklauncher.apps

import android.util.SparseArray
import java.util.*

enum class AppCategory(val code: Int) {
    OTHER(0),
    SETTINGS(1),
    VIDEO(2),
    MUSIC(3),
    GAME(4);

    companion object {
        private val intToType = SparseArray<AppCategory>()

        fun fromCategoryCode(code: Int): AppCategory {
            val category = intToType[code]
            return category ?: OTHER
        }

        fun fromName(name: String?): AppCategory? {
            return if (name == null)
                null
            else
                valueOf(name.uppercase(Locale.getDefault()).trim { it <= ' ' })
        }

        val VALUES = values()
        init {
            for (type in VALUES) {
                intToType.put(type.code, type)
            }
        }
    }
}